src/main/scala/nl.knaw.dans.easy.deposit/DataFiles.scala

```diff
@@ -61,6 +84,25 @@ case class DataFiles(bag: DansBag) extends DebugEnhancedLogging {
       .getOrElse(Failure(new Exception(s"no algorithm for ${ bag.baseDir }")))
   }
 
+  /**
+   * Lists information about a file.
+   *
+   * @param path a relative path into a data files.
+   * @return a [[FileInfo]] object
+   */
+  def fileInfo(path: Path = Paths.get("")): Try[FileInfo] = {
+    val manifestMap = bag.payloadManifests
+    val absolutePath = dataDir / path.toString
+    val fileExists = manifestMap.get(SHA1).orElse(manifestMap.values.headOption).get.contains(absolutePath)
+    if (fileExists) {
+      val checksum = manifestMap.get(SHA1).orElse(manifestMap.values.headOption) get absolutePath
```

Comment:
Questions:
* What will happen if there is no payload manifest at all?
* What are the invariants that we can count on?
* What do you report if an invariant is violated? 
* Would it make sense to check the file system to see if the file actually exists? What would you report if the file then exists on disk, but there is no manifest entry for it?

Comments:
* Lines 96 and 98 duplicate quite a lot.
* I see a lot of overlap with `fileInfoSeq`.

Reply:
In my understanding the bag is already validated. It is possible still to check whether `manifestMap` is `null`, and if it is so, to throw an exception, and take this exception in account in `DepositServlet`.  But I am wondering if that is a bit overkill, if the deposit indeed has already been validated?  The same applies to the question whether we would check if the  file exists in the filesystem, even if it is absent in the manifest file.

Reply:
The lines 96 and 98 were indeed very similar, thus I modified the code a bit